iceoryx_binding_c/include/iceoryx_binding_c/sleep_for.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_BINDING_C_SLEEP_FOR_H_
+#define IOX_BINDING_C_SLEEP_FOR_H_
+
+#ifdef _WIN32
```

Comment:
if we decide to keep this functionality, shouldn't `usleep` be moved to the windows platform?
We need to reduce platform dependent code in non platform parts of our codebase not increase it.